workflow for python, typescript SDK snippets

ccosnett opened the pull request:
now provides 3 different SDK testing workflows:

1.    `test_prod_snippets___prod_api_server___prod_sdk.yml`

This one tests everything in production.

--- 




2.    `test_prod_snippets___staging_api_server___latest_sdk.yml`

This workflow is simpler. It tests the production snippets generated by speakeasy at: https://spec.speakeasy.com/compasslabs/api/compass-api-with-code-samples
 using the **latest version of the SDK** with the  **staging api server**



---





3.    `test_prod_snippets___staging_api_server___release_candidate_sdk.yml`
   
This workflow is more complex. It tests the snippets at https://spec.speakeasy.com/compasslabs/api/compass-api-with-code-samples 

with the **release candidate version of the SDK** with the  **staging api server**.

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

scheuclu commented on .github/workflows/test_snippets.yml:
```diff
@@ -0,0 +1,111 @@
+
+jobs:
+
+  validate-environment:
+    runs-on: ubuntu-latest
+    outputs:
+      environment: ${{ steps.validate.outputs.environment }}
+    steps:
+      - name: Validate environment input
+        id: validate
+        run: |
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" || "${{ github.event_name }}" == "workflow_call" ]]; then
+            ENVIRONMENT="${{ inputs.environment }}"
+          else
+            ENVIRONMENT="local"
+          fi
+          if [[ "$ENVIRONMENT" != "staging" && "$ENVIRONMENT" != "prod" && "$ENVIRONMENT" != "local" ]]; then
+            echo "❌ Error: Invalid environment '$ENVIRONMENT'. Must be 'staging' or 'prod' or 'local'."
+            exit 1
+          fi
+          echo "✅ Environment '$ENVIRONMENT' is valid"
+          echo "environment=$ENVIRONMENT" >> $GITHUB_OUTPUT
```
Seems like this doesn't do anything?

scheuclu commented on .github/workflows/test_snippets.yml:
```diff
@@ -0,0 +1,111 @@
+    env:
+      COMPASS_API_KEY: ${{ secrets.COMPASS_API_KEY }}
+      # Provide a default if the upstream job isn't present:
+      UV_PACKAGE_VERSION: ${{ needs.set-versions.outputs.uv_package_version || 'latest' }}
```
What does `needs` refer to here?

ccosnett committed:
remove redundant file

ccosnett committed:
change filename

ccosnett commented on .github/workflows/test_snippets.yml:
```diff
@@ -0,0 +1,111 @@
+name: workflow 1
```
done

ccosnett committed:
simplify

ccosnett commented on v1/test_sdk_snippets/run_endpoint.py:
```diff
@@ -0,0 +1,64 @@
+    # Execute snippet in isolated globals
+    globs = {"__name__": "__main__"}
+    try:
+        exec(compile(code, filename=f"<sample:{endpoint}>", mode="exec"), globs, None)
```
@scheuclu not sure what you have in mind as an alternative. can we discuss?

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
wip

ccosnett committed:
wip

ccosnett committed:
wip

ccosnett committed:
wip

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
✨ Add SDK snippets jobs to main workflow

🚀 Integrate discover_sdk_snippets, run-sdk-snippets-python, and run-sdk-snippets-typescript jobs
🔧 Add environment validation and version management
📦 Support staging/prod/latest package versions
⚡ Matrix strategy for parallel endpoint testing
🎯 Conditional execution based on discovered endpoints

ccosnett committed:
COMMENT OUT JOBS

ccosnett committed:
🔢 Limit SDK snippets to first 10 endpoints for faster testing

scheuclu committed:
Debug

scheuclu committed:
Debug

scheuclu committed:
Debug

scheuclu committed:
Debug

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
WIP

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
STAGING

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
WIP

ccosnett committed:
remove v0 workflow

ccosnett committed:
running staging workflow

ccosnett committed:
STAGING_COMPASS_API_KEY

scheuclu reviewed (commented):
Aren't these workflows all almost identical? Can't we consolidate the logic?

scheuclu commented:
Is this still relevant @ccosnett
